### PR TITLE
Rewrite

### PR DESCRIPTION
**Pull request by ericadamski**
This is a large PR!

- update all async code to use Observables
- Rewrite UI using React

:bug: fixed

- The bugs referenced in #13  and #1 should be fixed now with a change to the redirect url

### CODE REVIEW

File: lib/authenticate.js [hunk 1]
```diff
@@ -15,119 +27,113 @@ const EXPIRES = `${THIS}:expires_at`;
 const ACCESS = `${THIS}:access_token`;
 const REFRESH = `${THIS}:refresh_token`;
 
-export class Authenticate extends EventEmitter {
-  constructor () {
-    super();
+const CLIENT = new Medium({
+    clientId: 'e789916f9247',
+    clientSecret: '6fd13cdbd973e79c92b161249928a41cd5c9e4e7',
```
Comment (frytyler) [inline lib/authenticate.js]: Do you want this exposed?

File: lib/authenticate.js [hunk 2]
```diff
@@ -15,119 +27,113 @@ const EXPIRES = `${THIS}:expires_at`;
...
+                              )
+                          );
+
+                          storeToken(token$);
+
+                          return token$;
+                      })
+                    : token
+        )
+        .mapTo(CLIENT);
```
Comment (frytyler) [inline lib/authenticate.js]: I stopped here lol. I'll come back to the review

File: lib/authenticate.js (hunk 1 above)
Comment (ericadamski) [reply lib/authenticate.js]: Yes

**Comment by ChrisChinchilla** [comment]
@ericadamski Not enough of a coder to comment, but can I check out this branch to test or does it rely on other PRs?

**Comment by ericadamski** [comment]
@ChrisChinchilla it is standalone! Feel free to test it :) I will get someone else to take a quick look at the code before I merge it!

File: lib/authenticate.js [hunk 3]
```diff
@@ -15,119 +27,113 @@ const EXPIRES = `${THIS}:expires_at`;
...
-  }
-};
+}
+
+export function authenticate() {
+    return hasToken()
+        .switchMap(
+            token =>
+                !token
+                    ? openBrowser().switchMap(code => {
```
Comment (frytyler) [inline lib/authenticate.js]: Is it worth moving this to a function that takes token. It's a bit hard to read in a ternary

**Pull request merged**